lib/ansible/modules/web_infrastructure/letsencrypt.py

```diff
@@ -315,12 +315,17 @@ def write_file(module, dest, content):
     has changed.
     '''
     changed = False
-    # create a tempfile with some test content
-    _, tmpsrc = tempfile.mkstemp()
+    # create a tempfile
+    fd, tmpsrc = tempfile.mkstemp()
+    os.close(fd)
     f = open(tmpsrc, 'wb')
```

Comment:
why not reuse `fd` instead of closing it and then opening it?

Reply:
I don't really want to clutter this code even more by working with `os.write`; it's a low-level API and you have to make sure the correct number of bytes is written by yourself; see https://docs.python.org/2/library/os.html#os.write

Reply:
That's fair, but you can wrap the file descriptor in a `file` object: https://docs.python.org/2/library/os.html#os.fdopen

Reply:
Ah, didn't know that one :) Now I'm using that.

Reply:
I'm also learning this as I go :smile: